src/ai/backend/manager/repositories/artifact/repository.py

```diff
@@ -113,6 +114,14 @@ def apply_entity_filters(
         if filters.source_registry_type is not None:
             conditions.append(ArtifactRow.source_registry_type == filters.source_registry_type)
 
+        # Handle availability filter
+        if filters.availability and len(filters.availability) > 0:
+            availability_values = [availability.value for availability in filters.availability]
+            if len(availability_values) == 1:
+                conditions.append(ArtifactRow.availability == availability_values[0])
+            else:
+                conditions.append(ArtifactRow.availability.in_(availability_values))
```

Comment:
It doesn't seem necessary to specifically compare and apply the number of items.